### PR TITLE
Remove Notary and Content Trust

### PR DESCRIPTION
**Pull request by dave-tucker**
**- What I did**
This commit removes Notary and Content Trust.
Notary v1 is due to be replaced with Notary v2 soon.
There is no clean migration path from one to the other.
For now, this removes all signing from LinuxKit.
We will look to add this back once a new Notary alternative
becomes available.

**- How I did it**
- Removed any command line flags relating to signing and content trust
- Removed `trust.go`
- Removed all references to signing and content trust from makefiles, scripts and documentation
- Ensured that everything still compiles

**- How to verify it**
If CI is still green, then all should be well.

**- Description for the changelog**

- Removed support for Docker Content Trust and Notary Signing

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

**Comment by deitch** [comment]
Hats off @dave-tucker  I will look early next week.

**Comment by rn** [comment]
Thanks. I'll take a look later as well. May I suggest that to merge this PR it requires approval from all active maintainers which currently seem to be @deitch @justincormack and myself to make sure we all (publicly) agree with this (@dave-tucker I assume your approval is implicit in making this PR in the first place)

**Comment by dave-tucker** [comment]
Yes @rn, my +1 is implicit

File: kernel/Makefile
Comment (deitch) [inline kernel/Makefile:141]: This line is odd.

File: src/cmd/linuxkit/moby/images.go
Comment (deitch) [inline src/cmd/linuxkit/moby/images.go:38]: We removed how this uses `trust`, but kept it as an arg. I think we should just get rid of it here, too.

File: src/cmd/linuxkit/build.go
Comment (deitch) [inline src/cmd/linuxkit/build.go:220]: Why do we pass `false` to trust, as opposed to just getting rid of it as an arg entirely?

File: src/cmd/linuxkit/cache/push.go
Comment (deitch) [inline src/cmd/linuxkit/cache/push.go:13]: We probably should remove `trust` as an arg here.

File: src/cmd/linuxkit/pkglib/docker.go
Comment (deitch) [inline src/cmd/linuxkit/pkglib/docker.go:23]: Why not just get rid of it?

**Review by deitch — CHANGES_REQUESTED** [review]
Two requests:

1. One of the comment changes is confusing
2. This PR removes the `trust` env var or arg from lots of places, but leaves it in a few places, always setting it to `false` or `""` and not passing it anywhere. We should just complete the process and get rid of those.

**Comment by dave-tucker** [comment]
@deitch good catched. most of it was some clumsy `sed` on my behalf :blush: I've gone through, fixed up the comments in question and removed the `trust bool` from `pkg/moby`.

**Review by deitch — DISMISSED** [review]
LGTM, and thank you for taking care of this.

We should wait for a LGTM from all of the core maintainers.

File: docs/packages.md
Comment (justincormack) [inline docs/packages.md:58]: Do we still need this?

File: scripts/push-manifest.sh
Comment (justincormack) [inline scripts/push-manifest.sh:71]: I think you can remove all the manifest-tool parts above as it is only used here

File: src/cmd/linuxkit/pkglib/build.go
Comment (justincormack) [inline src/cmd/linuxkit/pkglib/build.go:14]: thats a horrible name to give it!

File: tools/alpine/push-manifest.sh
Comment (justincormack) [inline tools/alpine/push-manifest.sh:5]: remove "signs it"

File: docs/packages.md
Comment (deitch) [reply docs/packages.md:58]: No, we do not. manifest-tool now is embedded as a library inside the linuxkit binary. We no longer exec out to it.

For that matter, what to we use expect, base64 and jq for?

File: tools/alpine/push-manifest.sh
Comment (justincormack) [inline tools/alpine/push-manifest.sh:12]: docker manifest did land a long time ago, can we use instead of manifest tool?

Comment (deitch) [reply tools/alpine/push-manifest.sh:12]: We should consider it as a replacement for the embedded library stuff (although I usually prefer embedded), but that definitely should be subject of a separate PR

File: docs/packages.md
Comment (rn) [reply docs/packages.md:58]: kernel build still uses this

File: docs/packages.md
Comment (rn) [inline docs/packages.md:183]: IIRC the size was only needed for signing. so can probably remove this section as well

File: kernel/Makefile
Comment (rn) [inline kernel/Makefile:146]: I wouldn't delete this comment yet, since it sounds like by removing content trust we can simplify the bits below

File: scripts/push-manifest.sh
Comment (rn) [reply scripts/push-manifest.sh:71]: we can probably remove this script. it;'s only used for kernel build and the alpine base. if we can replace the manifest tool with docker than there is no need for this script

File: kernel/Makefile
Comment (dave-tucker) [reply kernel/Makefile:146]: Addressed. Added a FIXME to the top to show that the comment isn't current anymore.

File: docs/packages.md
Comment (dave-tucker) [reply docs/packages.md]: Done

File: tools/alpine/push-manifest.sh
Comment (dave-tucker) [reply tools/alpine/push-manifest.sh]: Done

File: scripts/push-manifest.sh
Comment (dave-tucker) [reply scripts/push-manifest.sh:71]: I'll keep it here. We can remove it in a follow-up PR that removes manifest-tool and related bits

**Review by deitch — APPROVED** [review]

**Review by rn — APPROVED** [review]
Thanks for working on this.

**Review by justincormack — APPROVED** [review]

**Pull request merged**
